video_worker/generate_encode.py

```diff
@@ -113,99 +98,116 @@ def _call(self):
             mezz_parse_bitrate = self.VideoObject.mezz_bitrate.strip().split(' ')[0]
         else:
             mezz_parse_bitrate = None
-        """RESOLUTION as int"""
+
+        # RESOLUTION as int
         if self.VideoObject.mezz_resolution != 'Unparsed' and len(self.VideoObject.mezz_resolution) > 0:
             mezz_vert_resolution = int(self.VideoObject.mezz_resolution.strip().split('x')[1])
             mezz_horiz_resolution = int(self.VideoObject.mezz_resolution.strip().split('x')[0])
         else:
             mezz_vert_resolution = None
             mezz_horiz_resolution = None
-        """Aspect Ratio as float"""
+
+        # Aspect Ratio as float
         if mezz_vert_resolution is not None and mezz_horiz_resolution is not None:
             mezz_aspect_ratio = float(mezz_horiz_resolution) / float(mezz_vert_resolution)
         else:
             mezz_aspect_ratio = None
 
-        """Append commands"""
+        # Append commands
+
+        # TODO-1:
+        # if mezz is None then next condition in or works wich would attempt to check
+        # float(None) technically which is bound to raise TypeError
+        # float() argument must be a string or a number, not 'NoneType'
+        # I believe it should be an `and` condition like below
+        # if mezz_aspect_ratio is not None and float(mezz_aspect_ratio) == float(TARGET_ASPECT_RATIO):
+
         if mezz_aspect_ratio is not None or float(mezz_aspect_ratio) == float(TARGET_ASPECT_RATIO):
             aspect_fix = False
+
+        # TODO-1:
+        # This if condition would never be True because if above is false it means mezz_aspect_ratio is None
+        # which only happens if mezz_vert_resolution and mezz_horiz_resolution are None.
+
         elif mezz_vert_resolution == 1080 and mezz_horiz_resolution == 1440:
             aspect_fix = False
         else:
             aspect_fix = True
 
+        # TODO-1: This would raise TypeError when doing on  int(None) for mezz_vert_resolution.
         if int(self.EncodeObject.resolution) == int(mezz_vert_resolution):
             resolution_fix = False
         else:
             resolution_fix = True
 
-        if aspect_fix is False and resolution_fix is False:
-            return None
+        if not aspect_fix and not resolution_fix:
+            return
 
-        if aspect_fix is False and resolution_fix is True:
-            self.ffcommand.append("-vf")
-            self.ffcommand.append("scale=" + str(horiz_resolution) + ":" + str(self.EncodeObject.resolution))
+        ### MUSHI: COVERED UPTO THIS
 
-        elif aspect_fix is True:
+        if not aspect_fix and resolution_fix:
+            self.ffcommand.append('-vf')
+            self.ffcommand.append('scale=' + str(horiz_resolution) + ':' + str(self.EncodeObject.resolution))
+        # TODO-1: Flow would never reaches this line due to TypeError on mezz_aspect_ratio when it is None
```

Comment:
@mushtaqak If mezz_aspect_ratio is None, then this block shouldn't execute, it means we have a source file with an undetermined aspect ratio so we shouldn't attempt to make it conform to 16:9 -- you guys can definitely come up with a better solution here than mine, but that was the thinking here.